### PR TITLE
FB11569 Avatar island checkout brings empty tablet view

### PR DESCRIPTION
**Pull request by ElderOrb**
fix for https://highfidelity.fogbugz.com/f/cases/11569/Avatar-island-checkout-brings-empty-tablet-view

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6472/app_12218_PR.apk

File: libraries/ui/src/ui/OffscreenQmlSurface.cpp [hunk 1]
```diff
@@ -872,9 +872,20 @@ void OffscreenQmlSurface::loadInternal(const QUrl& qmlSource, bool createNewCont
...
+
+    // this ugly hack is attempt to simplify life of external script developers who could occasionally make path like this:
+    // var MARKETPLACE_WALLET_QML_PATH = Script.resourcesPath() + "qml/hifi/commerce/wallet/Wallet.qml";
+    // the logic below tries to rebuild url
+    static auto localResourceQmlUrlString = QUrl::fromLocalFile(PathUtils::resourcesPath() + "qml/").toString();
+    if (finalQmlSource.toString().startsWith(localResourceQmlUrlString, Qt::CaseInsensitive)) {
+        finalQmlSource = PathUtils::qmlBasePath() + finalQmlSource.toString().remove(0, localResourceQmlUrlString.length());
+    }
+
+    qDebug() << "qmlSource: " << finalQmlSource;
```
Comment (sethalves) [inline libraries/ui/src/ui/OffscreenQmlSurface.cpp]: let's remove these debug prints, if this is working.

File: libraries/ui/src/ui/OffscreenQmlSurface.cpp [hunk 2]
```diff
@@ -872,9 +872,20 @@ void OffscreenQmlSurface::loadInternal(const QUrl& qmlSource, bool createNewCont
...
+    if ((finalQmlSource.isRelative() && !finalQmlSource.isEmpty()) || finalQmlSource.scheme() == QLatin1String("file")) {
+        finalQmlSource = _qmlContext->resolvedUrl(finalQmlSource);
     }
+    qDebug() << "finalQmlSource qmlSource: " << finalQmlSource;
```
Comment (sethalves) [inline libraries/ui/src/ui/OffscreenQmlSurface.cpp]: debug print

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6472/HighFidelity-Beta-PR12218-dbbd55ea5719d05cf7a27eb0a1e172f157926fba.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6472/HighFidelity-Beta-PR12218-dbbd55ea5719d05cf7a27eb0a1e172f157926fba.exe

**Comment by ElderOrb** [comment]
closing this PR as opened one for master: https://github.com/highfidelity/hifi/pull/12219

**Pull request closed without merging**